aiida_common_workflows/workflows/relax/generator.py

```diff
@@ -68,28 +85,49 @@ def get_builder(
         threshold_forces=None,
         threshold_stress=None,
         previous_workchain=None,
+        is_insulator=False,
+        spin=SpinType.NONE,
+        initial_magnetization='auto',
```

Comment:
I tried to find the discussion on the decision to use a string as default and to mean auto-assign, but not sure if this is ideal. Apart from it mixing types (list and string) I think it could be simply inferred from the `spin` argument. I would set the default to `None` and just accept an explicit list as alternative. If it is `None` *and* `spin != SpinType.NONE` then the builder should automatically determine it. I think it never makes sense to have a spin calculation without setting an initial magnetization.

Reply:
Agreed